src/main/java/teammates/ui/webapi/output/CommentSearchResultData.java

```diff
@@ -0,0 +1,84 @@
+package teammates.ui.webapi.output;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import teammates.common.datatransfer.FeedbackResponseCommentSearchResultBundle;
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+
+/**
+ * API output format for a comment search result.
+ */
+public class CommentSearchResultData extends SessionResultsData {
+    private final FeedbackSessionData feedbackSession;
+
+    public CommentSearchResultData(FeedbackSessionAttributes session,
+                                   FeedbackResponseCommentSearchResultBundle bundle) {
+        super();
+        feedbackSession = new FeedbackSessionData(session);
+        bundle.questions.forEach((key, value) -> {
+            questions.addAll(buildQuestions(value, bundle));
+        });
+    }
+
+    private List<QuestionOutput> buildQuestions(List<FeedbackQuestionAttributes> questions,
+                                                FeedbackResponseCommentSearchResultBundle bundle) {
+        List<QuestionOutput> output = new ArrayList<>();
+        for (FeedbackQuestionAttributes question : questions) {
+            output.add(new QuestionOutput(question, buildResponses(bundle.responses.get(question.getId()), bundle)));
```

Comment:
Maybe we can break up this one-liner for better readability?

Reply:
Looks okay to me. Where would you suggest the breakpoint be?

Reply:
I felt output of `buildResponses()` can be separate out and saved as a variable? 

So that it is easier to see what are the params for `new QuestionOutput()`

Reply:
Yeap, I put the params of `new QuestionOutput()` to a new line :)